packages/ui/src/components/Designer/Sidebar/DetailView/BarCodePropEditor.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useContext } from 'react';
+import {
+  SchemaForUI,
+  DEFAULT_FONT_COLOR,
+  DEFAULT_BARCODE_BG_COLOR,
+  DEFAULT_BARCODE_COLOR,
+} from '@pdfme/common';
+import { I18nContext } from '../../../../contexts';
+import { SidebarProps } from '..';
+import { XMarkIcon } from '@heroicons/react/24/outline';
+
+const inputStyle = {
+  width: '90%',
+  color: '#333',
+  background: 'none',
+  borderRadius: 2,
+  border: '1px solid #767676',
+};
+
+const ColorInputSet = (props: {
```

Comment:
Just a note here:

This is very nearly a duplicate of the corresponding property in TextPropEditor. We considered moving this to shared code, but I know that this area will likely be reworked when Custom Schema Plugin support is implemented, so I thought it worth getting your opinion first @hand-dot.

Should we consolidate this to a shared file? Or leave them separate for now?

Reply:
Updated.

I think we should probably do the same for other form elements, but in a different PR